### PR TITLE
[WIP] 511 implementation using a service object

### PR DESCRIPTION
**Pull request by elg0nz**
* [x] Extract functionality into a Service object
* [x] Add test coverage by porting existing view tests to the Service Object tests
* [x] Test all the scenarios on the Tock #594 
* [x] Profit.

### CODE REVIEW

File: tock/hours/views.py [hunk 1]
```diff
@@ -313,6 +315,29 @@ def form_valid(self, form):
     def get_success_url(self):
         return reverse("ListReportingPeriods")
 
+def add_hours_view(request):
+    reporting_period = ReportingPeriod.objects.latest()
+    hours_adder = HoursAdder(
+        project_id = request.GET['project'],
+        hours = Decimal(request.GET['hours']),
```
Comment (gotascii) [inline tock/hours/views.py]: The Decimal-ing of `request.GET['hours']` should probs be handled in `HoursAdder` as it is a failure case we need to support (e.g. Someone submits the value `hi` for `request.GET['hours']`)

File: tock/hours/hours_adder.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            self.message = error_msg
+            self.operation_was_successful = False
+            return
+
+        if tco.hours_spent is None:
+            tco.hours_spent = 0
+
+        updated_hours = tco.hours_spent + self.hours
+        updated_hours = max(0, updated_hours)
+        tco.hours_spent = updated_hours
```
Comment (gotascii) [inline tock/hours/hours_adder.py:94]: This was my doing, but I think the above 2 lines could be combined w/o loosing clarity: `tco.hours_spent = max(0, updated_hours)`.

Comment (elg0nz) [reply tock/hours/hours_adder.py:94]: 👍

File: tock/hours/hours_adder.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    def __init__(self, **kwargs):
+        self.project_id = kwargs.get('project_id', None)
+        self.hours = kwargs.get('hours', None)
+        self.reporting_period_id = kwargs.get('reporting_period_id', None)
+        self.message = "Unknown error"
+        self.user_id = kwargs.get('user_id', None)
+        self.operation_was_successful = False
+
+    def generate_successful_message(self, hours, project):
```
Comment (gotascii) [inline tock/hours/hours_adder.py:26]: I feel like all this should maybe stay in the controller level (perhaps as another class) as it deals with creating urls and html. HoursAdder could just report success or failure and why in a non-web-based way. The controller should probably be in charge of then generating the url and html.

**Pull request closed without merging**